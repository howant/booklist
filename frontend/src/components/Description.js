import React, { Component } from "react";
import PropTypes from "prop-types";

class Description extends Component {
  render() {
    return (
      <div className="App-description">
        <h2>Title: {this.props.selectedBook.Title}</h2>
        <h2>Synopsis: {this.props.selectedBook.Synopsis}</h2>
      </div>
    );
  }
}

Description.propTypes = {
  selectedBook: PropTypes.shape({
    Title: PropTypes.string.isRequired,
    Synopsis: PropTypes.string.isRequired
  }).isRequired
};

export default Description;
