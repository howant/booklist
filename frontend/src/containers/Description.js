import { connect } from "react-redux";
import Description from "../components/Description";

const mapStateToProps = state => ({
  lists: state
});

const DescriptionContainer = connect(
  mapStateToProps,
)(Description);

export default DescriptionContainer;
