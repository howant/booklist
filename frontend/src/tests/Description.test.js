import React from 'react';
import Description from '../components/Description';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const lists = {
    selectedBook: {
      title: "Book One",
      synopsis: "synopsis goes here..."
    }
  }
  const tree = renderer
    .create(<Description selectedBook={lists}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});