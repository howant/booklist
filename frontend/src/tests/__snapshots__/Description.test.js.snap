// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App-description"
>
  <h2>
    Title: 
  </h2>
  <h2>
    Synopsis: 
  </h2>
</div>
`;
